internal/controllers/nmc_reconciler.go

```diff
@@ -169,6 +165,14 @@ func (r *NMCReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 		}
 	}
 
+	// removing label of loaded kmods
+	if !r.nodeAPI.IsNodeSchedulable(&node, nil) {
```

Comment:
We actually need to re-work the removal of the ready labels, since we cannot remove them for all the Modules, only for the ones without tolerations. But it is better if we do it in a different PR. I suggest we push this one , and then we will update the flow of the removing the ready labels. @TomerNewman will you have time to take care of this?

Reply:
Absolutely, after merging it I will open a new PR